xrpl/models/base_model.py

```diff
@@ -314,7 +374,7 @@ def to_dict(self: BaseModel) -> Dict[str, Any]:
             if getattr(self, key) is not None
         }
 
-    def _to_dict_elem(self: BaseModel, elem: Any) -> Any:
+    def _to_dict_elem(self: Self, elem: Any) -> Any:
         if isinstance(elem, BaseModel):
```

Comment:
are there any advantages to annotating the parameter `self: Self`, instead of `self: BaseModel` ? None of these functions are returning the `self` parameter, so I don't see the benefits of `typing.Self` class.

This was my take-away from the docs:
<img width="823" alt="image" src="https://github.com/XRPLF/xrpl-py/assets/21219765/4401075b-77d5-4e18-9a1f-246640ab1086">
<img width="823" alt="image" src="https://github.com/XRPLF/xrpl-py/assets/21219765/4401075b-77d5-4e18-9a1f-246640ab1086">

Reply:
ok